app/helpers/serializer_helper.rb

```diff
@@ -3,4 +3,13 @@ def ids_to_objs(ids)
     return [] if ids.blank?
     ids.map { |id| {id: id} }
   end
+
+  # Returns an array of the fields a serializer needs from it's object
+  # so we can #select only what the serializer will actually use
+  def required_attributes(model, serializer)
+    model_attributes = model.attribute_names
+    serializer_attributes = serializer._attributes.keys.map(&:to_s)
+
+    (serializer_attributes & model_attributes).map { |attr| "#{model.table_name}.#{attr}" }
```

Comment:
do you know how much of a performance impact this has?
I'd not do this if it does give us a big improvement. it's complicated.

because it is not easy, this method really needs unit tests...

Reply:
I'll add some tests. It seems to give ~10% improvement on maps.

Reply:
Spec added: https://github.com/openfoodfoundation/openfoodnetwork/pull/3833/commits/d9bafd56daaca7e98d030ae5205d15be7e8bcf55#diff-81148c208b0c1bdbfcd8e77f3fc82e57